OData/src/System.Web.OData/OData/Formatter/Serialization/ODataPrimitiveSerializer.cs

```diff
@@ -144,14 +144,26 @@ internal static object ConvertUnsupportedPrimitives(object value)
 
                     case TypeCode.DateTime:
                         DateTime dateTime = (DateTime)value;
-                        TimeZoneInfo timeZone = TimeZoneInfoHelper.TimeZone;
-                        if (dateTime.Kind == DateTimeKind.Utc || dateTime.Kind == DateTimeKind.Local)
+
+                        if (dateTime == DateTime.MinValue)
```

Comment:
Maybe you wanted to say if dateTime >= DateTime.MinValue + timeZone.BaseUtcOffset? 
If so, you will get ArgumentOutOfRangeException when timeZone.BaseUtcOffset is negative.

MinValue has always Kind = Unspecified (for DateTimeOffset property Offset set to zero). I believe we have to transfer it to client with zero offset.